Mathlib/Data/Complex/Basic.lean

```diff
@@ -0,0 +1,1382 @@
+/-
+Copyright (c) 2017 Kevin Buzzard. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard, Mario Carneiro
+
+! This file was ported from Lean 3 source module data.complex.basic
+! leanprover-community/mathlib commit 92ca63f0fb391a9ca5f22d2409a6080e786d99f7
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.Real.Sqrt
+
+/-!
+# The complex numbers
+
+The complex numbers are modelled as ℝ^2 in the obvious way and it is shown that they form a field
+of characteristic zero. The result that the complex numbers are algebraically closed, see
+`field_theory.algebraic_closure`.
+-/
+
+
+open BigOperators
+
+open Set Function
+
+/-! ### Definition and basic arithmmetic -/
+
+
+/-- Complex numbers consist of two `real`s: a real part `re` and an imaginary part `im`. -/
+structure Complex : Type where
+  re : ℝ
+  im : ℝ
+#align complex Complex
+
+-- Porting note: added as lemmas were not recognized as coercions
+attribute [coe] Complex.re
+attribute [coe] Complex.im
```

Comment:
We don't want these projections to be coercions, right?

Reply:
Thanks. Removed these.

They did seem wrong to me, but
I had put them in so `norm_cast` was recognized in at least some cases, but with the other approach of `ofReal'` with `coe` attribute these have no need.

Reply:
```suggestion
```